### PR TITLE
Develop into Master

### PR DESCRIPTION
**Pull request by jhkiehna**

### CODE REVIEW

**Comment by jhkiehna** [comment]
[ch618]

**Comment by jhkiehna** [comment]
[ch618]

File: js/api.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  if (request.contentScriptQuery == "fetchStories") {
+    chrome.storage.sync.get(["clubhouseToken"], item => {
+      $CLUBHOUSE_API_TOKEN = item.clubhouseToken;
+
+      fetch(
+        `${$CLUBHOUSE_API}/search/stories?token=${$CLUBHOUSE_API_TOKEN}&query=${
+          request.searchTerm
+        }`
+      )
+        .then(res => res.json())
```
Comment (audiolion) [inline js/api.js:13]: might be nice to do `.then(res => res.ok ? res : handleError)` before trying to call `.json`. `res.ok` is an alias for `res.status >= 200 && res.status < 300`. `handleError` would probably update your UI to say it couldnt fetch, clubhouse might be down or api token is invalid

File: js/script.js [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+ * When navigating back and forth in history, GitHub will preserve the DOM changes;
+ * This means that the old features will still be on the page and don't need to re-run.
+ * For this reason `onAjaxedPages` will only call its callback when a *new* page is loaded.
+ *
+ * Alternatively, use `onAjaxedPagesRaw` if your callback needs to be called at every page
+ * change (e.g. to "unmount" a feature / listener) regardless of of *newness* of the page.
+ */
+document.addEventListener(
+  "pjax:end",
+  () => {
```
Comment (audiolion) [inline js/script.js]: you are wrapping this in a function unnecessarily here, you could just do:

```js
  "pjax:end",
  injectButton,
```

File: js/script.js [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  searchInput.focus();
+
+  searchInput.addEventListener("blur", clearSearch, false);
+  searchInput.addEventListener("keypress", search, false);
+
+  searchButton.parentNode.removeChild(searchButton);
+};
+
+var injectButton = () => {
+  let commentTextArea = document.querySelector("#new_comment_field");
```
Comment (audiolion) [inline js/script.js]: would be nice to check for this on the pull request creation screen

File: js/script.js [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    `
+  );
+
+  let searchTerm = searchInput.value;
+
+  if (searchTerm) {
+    chrome.runtime.sendMessage(
+      { contentScriptQuery: "fetchStories", searchTerm: searchTerm },
+      response => {
+        response.data.forEach((story, index) => {
```
Comment (audiolion) [inline js/script.js]: this will loop through everything, it would be nice to just do `response.data.slice(0, 10).forEach`

File: js/script.js [hunk 5]
```diff
@@ -1,41 +1,16 @@
 let buttonContainer = document.createElement("div");
 buttonContainer.setAttribute("id", "clubhouse-button-container");
-buttonContainer.setAttribute(
```
Comment (audiolion) [inline js/script.js]: nice refactor!

**Comment by jhkiehna** [comment]
sfasafsadfddf
dsfas

**Comment by jhkiehna** [comment]
[ch1987]

**Comment by jhkiehna** [comment]
[ch706]

**Comment by jhkiehna** [comment]
[ch2105]

**Comment by jhkiehna** [comment]
afsdf

**Comment by jhkiehna** [comment]
[ch722]

File: js/script.js (hunk 5 above)
Comment (jhkiehna) [reply js/script.js]: thanks

**Pull request merged**